bigquery/google/cloud/bigquery/client.py

```diff
@@ -1327,18 +1320,18 @@ def copy_table(
             location = self.location
 
         job_ref = job._JobReference(job_id, project=project, location=location)
-
-        if isinstance(sources, str):
-            sources = TableReference.from_string(sources, default_project=self.project)
-
-        if isinstance(destination, str):
-            destination = TableReference.from_string(
-                destination, default_project=self.project
-            )
+        sources = _str_or_ref_to_table_ref(sources, default_project=self.project)
```

Comment:
Good idea. Done in https://github.com/googleapis/google-cloud-python/pull/7560/commits/3ab95d4ad7da8b2be27171716e294f024882536d. Except it's still a little more complicated than what you propose because `isinstance(sources, collections_abc.Sequence)` returns `True` when `sources` is a string.